switools/verifyswi.py

```diff
@@ -201,20 +207,55 @@ def verifySwi( swi, rootCA=ROOT_CA_FILE_NAME ):
    except X509CertException as e:
       return e.code
 
+def verifyAllSwi( swi, rootCA=ROOT_CA_FILE_NAME ):
+   # We will extract sub-images to /tmp to verify their signatures
+   workDir = "/tmp/verify-optims-%d" % os.getpid()
+   os.mkdir( workDir )
+   finalRetCode = VERIFY_SWI_RESULT.SUCCESS
+
+   try:
+      # Make sure the image we got is a swi file
+      if ( not os.path.isfile( swi ) or
+           os.system( "set -e; image=$(readlink -f %s); cd %s;"
+                      "unzip -o -q $image version" % ( swi, workDir ) ) ):
+         return VERIFY_SWI_RESULT.ERROR_NOT_A_SWI
+      optims = signaturelib.getOptimizations( swi, workDir )
+      # If image has multiple sub-images, verify each
+      if not ( optims is None or len( optims ) == 1 or "DEFAULT" in optims ):
+         print( "Optimizations in %s: %s" % ( swi, " ".join( optims ) ) )
+         signaturelib.extractSwadapt( swi, workDir )
+         for optim in optims:
+            optimImage = "%s/%s.swi" % ( workDir, optim )
+            # extract sub-image
+            os.system("%s/swadapt %s %s %s" % ( workDir, swi, optimImage, optim ) )
+            retCode = verifySwi( optimImage, rootCA )
+            os.remove( optimImage )
+            print( "%s: %s" % ( optim, VERIFY_SWI_MESSAGE[ retCode ] ) )
+            if retCode != VERIFY_SWI_RESULT.SUCCESS:
+               finalRetCode = VERIFY_SWI_RESULT.ERROR
+      # Finally check the container image (or legacy "one level" image)
+      retCode = verifySwi( swi, rootCA )
+      print( VERIFY_SWI_MESSAGE[ retCode ] )
+   finally:
+      shutil.rmtree( workDir )
```

Comment:
Not needed if the `try` above is removed.

Reply:
changed in last push